src/FilterLists.Api/Program.cs

```diff
@@ -1,12 +1,23 @@
 ﻿using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using FilterLists.Api.DependencyInjection.Extensions;
+using FilterLists.Data.Seed.Extensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace FilterLists.Api
 {
     public static class Program
     {
-        public static void Main(string[] args) => CreateWebHostBuilder(args).Build().Run();
+        public static void Main(string[] args)
+        {
+            CreateWebHostBuilder(args).Build().MigrateDbContext<Data.FilterListsDbContext>((context,service)=>
+            {
 
+                var dataPath = service.GetService<IConfiguration>()["DataDirectory:Path"].ToString();
+                new SeedFilterListsDbContext().SeedOrUpdateAsync(context,dataPath).Wait();
```

Comment:
Synchronously blocking until the task completes via `.Wait()` seems to negate many of the benefits of `async`. Is it possible to use `await` to allow `Main()` to continue on while seeding completes?

Reply:
[Follow this document](https://msdn.microsoft.com/en-us/magazine/jj991977.aspx). I use Wait() to make this task run Synchronously. SeedOrUpdateAsync() must complete because the Main method can’t be async. If the Main method were async, it could return before it completed, causing the program to end. In SeedOrUpdateAsync(), other method can run with async.